src/main/java/it/pagopa/transactions/client/WalletClient.java

```diff
@@ -0,0 +1,45 @@
+package it.pagopa.transactions.client;
+
+import it.pagopa.generated.wallet.v1.api.WalletsApi;
+import it.pagopa.generated.wallet.v1.dto.WalletInfoDto;
+import it.pagopa.transactions.exceptions.InvalidRequestException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClientResponseException;
+import reactor.core.publisher.Mono;
+
+import java.util.UUID;
+
+@Component
+@Slf4j
+public class WalletClient {
+
+    @Autowired
+    @Qualifier("walletWebClient")
+    private WalletsApi walletWebClient;
+
+    public Mono<WalletInfoDto> getWalletInfo(
+                                             String walletId
+
+    ) {
+        return walletWebClient
+                .getWalletAuthDataById(UUID.fromString(walletId))
+                .doOnError(
+                        WebClientResponseException.class,
+                        WalletClient::logWebClientException
+                )
+                .onErrorMap(
+                        err -> new InvalidRequestException("Error while invoke method for retrieve wallet info")
```

Comment:
This exception is mapped to 400 http error code, maybe a bad gateway is more appropriate. what do you think?

Reply:
Fixed with 11a7830b81e217048b258f83901712cdb1d4a68a